### DISCUSSION TITLE
requestPolicyExchange type error

### DISCUSSION THREAD

**Discussion by gonta1026** [urql-graphql/urql#3466] [Q&A] [answered]
I get an error when I put requestPolicyExchange into exchange.
Is this a library version issue?

```tsx
const newUrqlClient = ({ url }: { url: string }) => {
  return new Client({
    fetch,
    url,
    exchanges: [cacheExchange, devtoolsExchange, fetchExchange, requestPolicyExchange({ shouldUpgrade: () => true })],
    requestPolicy: 'network-only',
    fetchOptions: {
      headers: {
        Accept: '*/*',
        'Content-Type': 'application/json',
      },
    },
  })
}
```

### screen shot
<img width="1318" alt="スクリーンショット 2023-12-14 17 02 20" src="https://github.com/urql-graphql/urql/assets/45672476/ec480fe0-1af7-4cbd-88ec-5a09974bc16a">

### version
- "urql": "^4.0.4",
- "@urql/core": "^4.0.10",
- "@urql/exchange-request-policy": "^1.0.2",
- "graphql": "^16.7.1",
- "typescript": "5.2.2",

**Comment by JoviDeCroock** [comment, marked as answer]
This is most often cause by duplication in your dependencies, `npx yarn-deduplicate` or `npm dedupe` fix these.

Just saying but in `urql` synchronous exchanges come before asynchronous ones, in your current chain of exchanges your requestPolicyExchange does nothing while `[devtoolsExchange, requestPolicyExchange({ shouldUpgrade: () => true }), cacheExchange, fetchExchange]` would fix that

**Reply by gonta1026** [reply]
> This is most often cause by duplication in your dependencies, npx yarn-deduplicate or npm dedupe fix these.

In my case, these did not solve the problem, but I was able to solve it by deleting yarn.lock and node_modules and doing yarn install again.

> Just saying but in urql synchronous exchanges come before asynchronous ones, in your current chain of exchanges your requestPolicyExchange does nothing while [devtoolsExchange, requestPolicyExchange({ shouldUpgrade: () => true }), cacheExchange, fetchExchange] would fix that

Thanks for this as well. I would like to review the documentation as I did not check the documentation enough.

https://formidable.com/open-source/urql/docs/advanced/authoring-exchanges/#synchronous-first-asynchronous-last
